doc/source/cluster/kubernetes/user-guides/resource-isolation-with-writable-cgroups.md

```diff
@@ -0,0 +1,187 @@
+(resource-isolation-with-writable-cgroups)=
+
+# Resource Isolation with Kubernetes Writable Cgroups
+
+This guide covers how to enable Ray resource isolation on GKE using [writable cgroups](https://docs.cloud.google.com/kubernetes-engine/docs/how-to/writable-cgroups).
+Ray resource isolation (introduced in v2.51.0) significantly improves Ray's reliability by using cgroupsv2 to reserve dedicated CPU and memory
+resources for critical system processes. Historically, enabling resource isolation required privileged containers capable of writing to
+the `/sys/fs/cgroup` file system. This approach was not recommended due to the security risks associated with privileged containers. In newer versions of GKE,
+you can enable writable cgroups, granting containers read-write access to the cgroups API without requiring privileged mode.
+
+## Create a GKE Cluster with writable cgroups enabled
+
+To use Ray resource isolation on Kubernetes without privileged containers, you must use a platform that supports cgroupsv2 and writable cgroups.
+
+On GKE, create a cluster with writable cgroups enabled as follows:
+```bash
+$ cat > containerd_config.yaml << EOF
+writableCgroups:
+  enabled: true
+EOF
+
+$ gcloud container clusters create ray-resource-isolation \
+    --cluster-version=1.34 \
+    --machine-type=e2-standard-16 \
+    --num-nodes=3 \
+    --scopes="cloud-platform" \
```

Comment:
Is this required for Ray on GKE? It's not required for the writable cgroups feature

Reply:
It's referenced in the general guide for updating containerd config https://docs.cloud.google.com/kubernetes-engine/docs/how-to/customize-containerd-configuration#apply-containerd-configuration-to-new-clusters